extensions/azurecore/src/account-provider/auths/azureDeviceCode.ts

```diff
@@ -57,6 +57,11 @@ export class AzureDeviceCode extends AzureAuth {
 
 	}
 
+	public async promptForConsent(resourceId: string): Promise<undefined> {
+		vscode.window.showErrorMessage(localize('azure.deviceCodeDoesNotSupportConsent', "Device code authentication does not support prompting for consent. Switch the authentication method in settings to code grant."));
```

Comment:
Maybe explain a bit here (or link to a doc) why we need to prompt for consent here so the user doesn't blame us for it 😄

Reply:
Yeah I actually need to do some work in making this less abrupt, right now it could literally just popup in the middle of their work and start opening a browser which can throw people off